### PR TITLE
added publish content

### PR DESCRIPTION
**Pull request by bradygaster**
Added VS publish content. cc @IEvangelist @davidfowl @timheuer 

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/whats-new/preview-4.md](https://github.com/dotnet/docs-aspire/blob/8a8b1f55f89af5af3b5ecc458edbc5780ff86a18/docs/whats-new/preview-4.md) | [.NET Aspire preview 4](https://review.learn.microsoft.com/en-us/dotnet/aspire/whats-new/preview-4?branch=pr-en-us-477) |

### CODE REVIEW

File: docs/whats-new/preview-4.md [hunk 1]
```diff
@@ -604,15 +604,21 @@ Tool authors can support this very small set of resource types to model lots of
 
 We've deprecated the abstract resource types that were supported in previous versions. Deployment tools may still support those resource types, but they are no longer part of the core manifest schema.
 
+## Azure Developer CLI (azd)
+
+The [Azure Developer CLI](https://aka.ms/azd) (azd) has been updated to support the new manifest types introduced in preview 4. Azd supports prompting for parameter resources, and supports the bicep resource which enables deploying any thing. All Azure resources can be described using a bicep module.
+
+In addition to new features, azd will automatically create secrets in Azure Container Apps for any parameter marked as a secret, and any environment variable that references a secret or uses a connection string.
+
 ### Visual Studio Publish to Azure
```
Comment (timheuer) [inline docs/whats-new/preview-4.md:613]: Can we change this to ## to put on same level as azd and better TOC at the top?

File: docs/whats-new/preview-4.md [hunk 2]
```diff
@@ -604,15 +604,21 @@ Tool authors can support this very small set of resource types to model lots of
...
+![.NET Aspire right-click publish in Visual Studio](media/preview-4/aspire-azd-publish.gif)
 
-In addition to new features, azd will automatically create secrets in Azure Container Apps for any parameter marked as a secret, and any environment variable that references a secret or uses a connection string.
+
+With so many similarities between Container Apps' support for [add-on services](https://learn.microsoft.com/azure/container-apps/services) like PostgreSQL, Kafka, and Redis and the support offered by mirroring .NET Aspire components ([PostgreSQL](https://learn.microsoft.com/dotnet/aspire/database/postgresql-component?tabs=dotnet-cli), [Kafka](https://learn.microsoft.com/dotnet/aspire/messaging/kafka-component?tabs=dotnet-cli), and [Redis](https://learn.microsoft.com/dotnet/aspire/caching/stackexchange-redis-component?tabs=dotnet-cli), for example), targeting Azure Container Apps as our **first** publishing destination was an easy decision. You'll be able to use Azure Container Apps as your development publishing target out-of-the-box using Visual Studio Preview's upcoming 17.10 preview 3 release. 
 
 ### Prompting for parameters
```
Comment (timheuer) [inline docs/whats-new/preview-4.md:623]: @bradygaster I think given the VS experience doesn't support inputs for Parameter in this preview, maybe a remark in this section?

File: docs/whats-new/preview-4.md [hunk 3]
```diff
@@ -604,15 +604,21 @@ Tool authors can support this very small set of resource types to model lots of
...
+
+In addition to new features, azd will automatically create secrets in Azure Container Apps for any parameter marked as a secret, and any environment variable that references a secret or uses a connection string.
+
 ### Visual Studio Publish to Azure
 
-This release we've enabled a Visual Studio publish experience for .NET Aspire applications.
+This release we've enabled a Visual Studio publish experience for .NET Aspire applications. This new deployment experience enables a single right-click-publish gesture that results in *all* of the nodes of your .NET Aspire project being published as individual Azure Container Apps in an Azure Container Apps Environment. We've chosen Azure Container Apps as the premier container PaaS destination for .NET Aspire apps as it enables orchestrated containerized deployments of multi-node microservice and distributed monolith applications. Even better - the publishing experience in Visual Studio uses `azd` under the hood, so you'll be able to publish .NET Aspire apps directly from Visual Studio to any `azd` environment you have running in Azure.
```
Comment (timheuer) [inline docs/whats-new/preview-4.md:615]: "...being published as individual Azure Container Apps in an Azure Container Apps Environment, ideal for dev/test scenarios."

Add the "ideal for dev/test scenarios"?

**Review by timheuer — CHANGES_REQUESTED** [review]

File: docs/whats-new/preview-4.md (hunk 2 above)
Comment (bradygaster) [reply docs/whats-new/preview-4.md:623]: good point - i need to mention the AZD auth login part too. thanks!

Comment (timheuer) [reply docs/whats-new/preview-4.md:607]: lgtm

**Review by timheuer — APPROVED** [review]

**Comment by gkulin** [comment]
is there an aka link that will point to this once it's published? Want to make sure I include it in azd's 1.7.0 release blog

**Review by IEvangelist — APPROVED** [review]

File: docs/whats-new/preview-4.md [hunk 4]
```diff
@@ -687,3 +677,23 @@ If the secret is referenced directly, the Container App secret will be a referen
...
+
+In addition to new features, azd will automatically create secrets in Azure Container Apps for any parameter marked as a secret, and any environment variable that references a secret or uses a connection string.
+
+## Visual Studio Publish to Azure
+
+This release we've enabled a Visual Studio publish experience for .NET Aspire applications. This new deployment experience enables a single right-click-publish gesture that results in *all* of the nodes of your .NET Aspire project being published as individual Azure Container Apps in an Azure Container Apps Environment, ideal for dev/test scenarios. Even better - the publishing experience in Visual Studio uses `azd` under the hood, so you'll be able to publish .NET Aspire apps directly from Visual Studio to any `azd` environment you have running in Azure.
+
+![.NET Aspire right-click publish in Visual Studio](media/preview-4/aspire-azd-publish.gif)
```
Comment (IEvangelist) [inline docs/whats-new/preview-4.md:691]:
```suggestion
:::image type="content" source="media/preview-4/aspire-azd-publish.gif" lightbox="media/preview-4/aspire-azd-publish.gif" alt-text=".NET Aspire right-click publish in Visual Studio":::
```

File: docs/whats-new/preview-4.md [hunk 5]
```diff
@@ -687,3 +677,23 @@ If the secret is referenced directly, the Container App secret will be a referen
...
+
+:::image type="content" source="media/preview-4/aspire-azd-publish.gif" lightbox="media/preview-4/aspire-azd-publish.gif" alt-text=".NET Aspire right-click publish in Visual Studio":::
+
+With so many similarities between Container Apps' support for [add-on services](https://learn.microsoft.com/azure/container-apps/services) like PostgreSQL, Kafka, and Redis and the support offered by mirroring .NET Aspire components ([PostgreSQL](https://learn.microsoft.com/dotnet/aspire/database/postgresql-component?tabs=dotnet-cli), [Kafka](https://learn.microsoft.com/dotnet/aspire/messaging/kafka-component?tabs=dotnet-cli), and [Redis](https://learn.microsoft.com/dotnet/aspire/caching/stackexchange-redis-component?tabs=dotnet-cli), for example), targeting Azure Container Apps as our **first** publishing destination was an easy decision. You'll be able to use Azure Container Apps as your development publishing target out-of-the-box using Visual Studio Preview's upcoming 17.10 preview 3 release.
```
Comment (IEvangelist) [inline docs/whats-new/preview-4.md:693]:
```suggestion
With so many similarities between Container Apps' support for [add-on services](/azure/container-apps/services) like PostgreSQL, Kafka, and Redis and the support offered by mirroring .NET Aspire components ([PostgreSQL](/dotnet/aspire/database/postgresql-component?tabs=dotnet-cli), [Kafka](/dotnet/aspire/messaging/kafka-component?tabs=dotnet-cli), and [Redis](/dotnet/aspire/caching/stackexchange-redis-component?tabs=dotnet-cli), for example), targeting Azure Container Apps as our **first** publishing destination was an easy decision. You'll be able to use Azure Container Apps as your development publishing target out-of-the-box using Visual Studio Preview's upcoming 17.10 preview 3 release.
```

**Pull request merged**